<launch>

    <arg name="frame_id" default="ground"/>
    <arg name="base_frame_id" default="/base_link"/>
    
    <arg name="height_map" default="true"/>
    <arg name="sensor_model/max_range" default="12"/>
    <arg name="latch" default="false"/>
    <arg name="resolution"  default="0.05"/>

    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server"> 
        
        <remap from="/cloud_in" to="/laser_cloud_map"/>
        <remap from="/projected_map" to="/map"/>
        <param name="frame_id"    value="$(arg frame_id)" />
        <param name="base_frame_id"    value="$(arg base_frame_id)" />
        <param name="height_map"    value="$(arg height_map)" />
        <param name="sensor_model/max_range"    value="$(arg sensor_model/max_range)" />
        <param name="latch" value="$(arg latch)"/>
        <param name="resolution"    value="$(arg resolution)"/>
        <param name="occupancy_min_z"   value="-1"/>
        <param name="occupancy_max_z"   value="6"/>
        <param name="occupancy_2d_min_z"   value="0.0"/>
        <param name="occupancy_2d_max_z"   value="1.0"/>
        <param name="filter_ground"	value="false"/>
        <!-- <param name="ground_filter/angle"	value="0.3"/> -->
        <!-- <param name="ground_filter/plane_distance"	value="0.2"/> -->
        <!-- <param name="ground_filter/distance"	value="0.1"/> -->
        <param name="pointcloud_min_z"   value="-1"/>
        <param name="pointcloud_max_z"   value="6"/>
    </node>
<!-- 
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server_projection" output="screen">
        
        <remap from="/cloud_in" to="/laser_cloud_surround"/>
        <remap from="/octomap_full" to="/octomap_full_p"/>
        <remap from="/octomap_binary" to="/octomap_binary_p"/>
        
        <remap from="/occupied_cells_vis_array" to="/occupied_cells_vis_array_p"/>

        <remap from="/octomap_point_cloud_centers" to="/octomap_point_cloud_centers_p"/>
        <remap from="/octomap_collision_object" to="/octomap_collision_object_p"/>
        <remap from="/collision_map_out" to="/collision_map_out_p"/>
        <remap from="/projected_map" to="/p_map"/>

        <param name="frame_id"    value="$(arg frame_id)" />
        <param name="base_frame_id"    value="$(arg base_frame_id)" />
        <param name="height_map"    value="$(arg height_map)" />
        <param name="sensor_model/max_range"    value="$(arg sensor_model/max_range)" />
        <param name="latch" value="$(arg latch)"/>
        <param name="resolution"    value="$(arg resolution)"/>
        <param name="occupancy_min_z"   value="0.2"/>
        <param name="occupancy_max_z"   value="0.7"/>
        <param name="filter_ground"	value="false    "/>
        <param name="ground_filter/angle"	value="0.2"/>
        <param name="ground_filter/plane_distance"	value="0.20"/>
        <param name="ground_filter/distance"	value="0.20 "/>
        <param name="pointcloud_min_z"   value="0.2"/>
        <param name="pointcloud_max_z"   value="0.7"/>
    </node>
 -->

</launch>
